hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java

```diff
@@ -154,7 +154,7 @@ private long doCopy(CopyListingFileStatus source, Path target,
       if (!source.isSplit()) {
         DistCpUtils.compareFileLengthsAndChecksums(source.getLen(), sourceFS,
                 sourcePath, sourceChecksum, targetFS,
-                targetPath, skipCrc, source.getLen());
+                targetPath, skipCrc, bytesRead);
```

Comment:
The test failures are related to the change. Probably we need to add offset.
```suggestion
                targetPath, skipCrc, offset + bytesRead);
```

Reply:
"
      compareFileLengths(sourceFileStatus, targetPath, configuration, bytesRead
          + offset);
"
When I copied from old version hadoop, I encountered a line break...Thank you